frontend/frontend/target_systems/_grpc.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from typing import Optional, cast
+import grpc
+import target_system_provider.target_system_provider_pb2_grpc as tsp
+import target_system_provider.target_system_provider_pb2 as messages
+from ._interface import TargetSystem, TargetSystemProvider
+
+logger = logging.getLogger(__file__)
+
+
+class _GrpcTargetSystem(TargetSystem):
+    """Internal gRPC implementation of TargetSystem"""
+
+    target_id: str
+
+    def __init__(self, target_id: str, address: str, port: int) -> None:
+        self.target_id = target_id
+        self.address = address
+        self.port = port
+
+
+class _GrpcTargetSystemProvider(TargetSystemProvider):
+    """TargetSystemProvider implementation using the target_system_provider gRPC protocol
+    for forwarding of requests to a remote service."""
+
+    channel: Optional[grpc.Channel]
+
+    def __init__(self, server_address: str) -> None:
+        """Initalizes a new TargetSystemProvider connecting to a given server address.
+        :param server_address: The address of a gRPC server.
+        """
+        super().__init__()
+        self.channel = grpc.insecure_channel(server_address)  # TODO: TLS
+
+    def close_channel(self):
+        """Closes the underlying gRPC channel"""
+        self.channel.close()
```

Comment:
Calling this twice will result in an exception since `self.channel` will be `None` the second time. If it's only meant to be called through `create_grpc_target_system_provider` it might be nice to indicate the method as private and to make sure it isn't `None`